server/src/main/java/org/opensearch/gateway/BaseGatewayShardAllocator.java

```diff
@@ -74,7 +78,45 @@ public void allocateUnassigned(
         ExistingShardsAllocator.UnassignedAllocationHandler unassignedAllocationHandler
     ) {
         final AllocateUnassignedDecision allocateUnassignedDecision = makeAllocationDecision(shardRouting, allocation, logger);
+        executeDecision(shardRouting, allocateUnassignedDecision, allocation, unassignedAllocationHandler);
+    }
+
+    /**
+     * Allocate Batch of unassigned shard  to nodes where valid copies of the shard already exists
+     * @param shardRoutings the shards to allocate
+     * @param allocation the allocation state container object
+     */
+    public void allocateUnassignedBatch(List<ShardRouting> shardRoutings, RoutingAllocation allocation) {
+        // make Allocation Decisions for all shards
+        HashMap<ShardRouting, AllocateUnassignedDecision> decisionMap = makeAllocationDecision(shardRoutings, allocation, logger);
+        assert shardRoutings.size() == decisionMap.size() : "make allocation decision didn't return allocation decision for "
+            + "some shards";
+        // get all unassigned shards iterator
+        RoutingNodes.UnassignedShards.UnassignedIterator iterator = allocation.routingNodes().unassigned().iterator();
+
+        while (iterator.hasNext()) {
+            ShardRouting shard = iterator.next();
+            try {
+                if (decisionMap.isEmpty() == false) {
+                    if (decisionMap.containsKey(shard)) {
+                        executeDecision(shard, decisionMap.remove(shard), allocation, iterator);
+                    }
+                } else {
+                    // no need to keep iterating the unassigned shards, if we don't have anything in decision map
+                    break;
+                }
+            } catch (Exception e) {
+                logger.error("Failed to execute decision for shard {} while initializing {}", shard, e);
```

Comment:
why are we catching exception here? During single shard allocation as well, we don't catch exception, here as well it should be same.

Reply:
It's just for debugging purposes. We don't expect failures, so if anything fails, it'll log those errors with custom message to catch it easily in logs.

Reply:
then you should rethrow, it is getting swallowed right now

Reply:
Only reason I remember we did it to keep executing decisions for rest of the shards.

I see the existing behaviour from AllocationService is to throw and exit the flow even if a single shard returns failure during executeDecision.

I'll re-throw to keep the same behaviour.